src/price/PriceHelper.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace hiqdev\php\billing\price;
+
+use Money\Currencies\ISOCurrencies;
+use Money\Currency;
+use Money\Money;
+use Money\Parser\DecimalMoneyParser;
+
+class PriceHelper
+{
+    public static function buildMoney(string $price, string $currency): ?Money
+    {
+        if (!is_numeric($price)) {
+            throw new \InvalidArgumentException('price of the Money must be numeric');
+        }
+        if (!self::checkFloat($price)) {
+            return new Money($price, new Currency($currency));
+        }
+        if (!is_int($price) && ($price * 100) == ((int) ($price * 100))) {
+            return (new DecimalMoneyParser(new ISOCurrencies()))->parse($price, new Currency($currency));
```

Comment:
DecimalMoneyParser is a service, don't create services with `new` keyword in a code: use DI